### PR TITLE
Integration testing for sigma autodiscover functionality

### LINKED ISSUE #41: Add unit test for pySigma autodiscover functionality
Introduce a new unit test that attempts to use pySigma's autodiscovery functionality and ensures that this backend can be automatically discovered and installed correctly.

### PR DESCRIPTION
**Pull request by kelnage**
A simple integration test to help ensure that this backend and its associated pipelines are picked up correctly by the pySigma library.

@mostafa hope you're happy to take a look at this - you're probably the most familiar with the auto-discovery functionality in pySigma, so I feel like you'll likely catch any issues that might crop up!

Fixes #41

### CODE REVIEW

File: tests/test_pysigma_integration.py [hunk 1]
```diff
@@ -0,0 +1,31 @@
...
+
+
+@pytest.fixture
+def installed() -> InstalledSigmaPlugins:
+    return InstalledSigmaPlugins.autodiscover()
+
+
+def test_auto_discover_loki_be(installed: InstalledSigmaPlugins):
+    assert "loki" in installed.backends
+    assert installed.backends["loki"] is LogQLBackend
```
Comment (mostafa) [inline tests/test_pysigma_integration.py:18]: Better to use `isinstance` for assertion of type here:
```suggestion
    assert isinstance(installed.backends["loki"], LogQLBackend)
```

File: tests/test_pysigma_integration.py [hunk 2]
```diff
@@ -0,0 +1,31 @@
...
+
+
+def test_auto_discover_loki_pipelines(installed: InstalledSigmaPlugins):
+    pipelines = [
+        loki_grafana_logfmt,
+        loki_promtail_sysmon,
+        loki_okta_system_log,
+    ]
+    print(installed.pipelines)
```
Comment (mostafa) [inline tests/test_pysigma_integration.py:27]: Remove `print`:
```suggestion
```

File: tests/test_pysigma_integration.py [hunk 3]
```diff
@@ -0,0 +1,31 @@
...
+    for pipeline in pipelines:
+        name = pipeline.__name__
+        assert name in installed.pipelines
+        assert installed.pipelines[name] == pipeline
```
Comment (mostafa) [inline tests/test_pysigma_integration.py:31]: This is a nitpick. Feel free to disregard!
```suggestion
    loki_pipelines = [
        loki_grafana_logfmt,
        loki_promtail_sysmon,
        loki_okta_system_log,
    ]
    for pipeline in loki_pipelines:
        assert pipeline.__name__ in installed.pipelines
        assert installed.pipelines[pipeline.__name__] == pipeline
```

**Review by mostafa — COMMENTED** [review]
I don't think changing class names has any effect on the autodiscovery feature. Plus, I am changing it to be become more automated, as you already know.
I left a few comments and a nit.

File: tests/test_pysigma_integration.py (hunk 1 above)
Comment (kelnage) [reply tests/test_pysigma_integration.py:18]: I don't think it is better to use `isinstance` here. `installed.backends["loki"]` doesn't contain an instance of the LogQLBackend - it contains the class - and `isinstance(installed.backends["loki"], LogQLBackend)` correctly returns false.

File: tests/test_pysigma_integration.py (hunk 3 above)
Comment (kelnage) [reply tests/test_pysigma_integration.py:31]: More than happy to rename the list to `loki_pipelines`, but why the suggestion against the use of the `name` local variable?

Comment (mostafa) [reply tests/test_pysigma_integration.py:31]: It is pythonic to use `.__name__`, unless you are eager to shorten it with `name`.

**Pull request merged**